fastcrypto/src/secp256k1/recoverable.rs

```diff
@@ -295,6 +305,35 @@ impl Default for Secp256k1RecoverableSignature {
     }
 }
 
+impl Secp256k1RecoverableSignature {
```

Comment:
it's the same code as for R1, can it be a generic or a trait like
TryFromNonRecoverable {
type sig;
type pk;
 pub fn try_from_nonrecoverable () ...
 }

Reply:
There's a lot of duplicated code between the two already, so I'd rather fix that in another PR.

Reply:
I've added a new issue #393